### PR TITLE
Sign-out user when id is not found in database

### LINKED ISSUE #883: Invalid user sessions improperly handled on the delete service worker branch
## Describe the bug

**On drop service worker branch:**

When opening the website with an invalid user session (such as the session of a user that doesn't exist anymore), we are left on a completely blank page (nothing, no headers, no UI to log in/out)

This issue is possibly also present in other situations related to user sessions and rights.

## To Reproduce

1. Have a working logged in user session
2. Delete user from database, or use a fresh new test database
3. Go to any page of the website

## Expected behavior

User session should be deleted to allow signing in/up

## Additional information

The query that fails is `UserWithWorkspacesQuery`.
The error that I got was `{"errors":[{"message":"Couldn't find an user with id: \"8747d5b2-8861-4ff7-bf43-7e12bcd56860\"","locations":[{"line":2,"column":3}],"path":["user"]}],"data":null}`

### PR DESCRIPTION
**Pull request by Jordanlelay**
## Work performed

- Fix user not disconnected when not found in database

## Resolved issues

Closes #883

### CODE REVIEW

File: typescript/web/src/hooks/use-user.tsx [hunk 1]
```diff
@@ -52,10 +52,14 @@ const useUserQuery = <TTypes extends UserTupleTypes>(
 ): TTypes[2] | undefined => {
   const id = useUserId() ?? "";
   const skip = isEmpty(id);
-  const { data } = useQuery<TTypes[0], TTypes[1]>(query, {
+  const { data, error } = useQuery<TTypes[0], TTypes[1]>(query, {
     variables: { id },
     skip,
   });
+
+  if (error) {
```
Comment (bbenoist) [inline typescript/web/src/hooks/use-user.tsx:60]: The handling should be more selective and keep the other errors follow the usual procedure.

**Review by bbenoist — CHANGES_REQUESTED** [review]

**Review by bbenoist — CHANGES_REQUESTED** [review]

**Review by bbenoist — APPROVED** [review]
Code looks good to me 👍  Placed at the correct location!

**Comment by bbenoist** [comment]
@valerian is doing functional tests and will approve and eventually merge the PR if OK 🚀

**Comment by nicOwlas** [comment]
Functional test works perfectly 👍

**Review by valerian — APPROVED** [review]

**Pull request merged**